digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/SpeedLimitService.scala

```diff
@@ -344,12 +328,12 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
     * Saves speed limit values when speed limit is split to two parts in UI (scissors icon). Used by Digiroad2Api /speedlimits/:speedLimitId/split POST endpoint.
     */
   def split(id: Long, splitMeasure: Double, existingValue: Int, createdValue: Int, username: String, municipalityValidation: (Int) => Unit): Seq[SpeedLimit] = {
-    withDynTransaction {
-      val newId = dao.splitSpeedLimit(id, splitMeasure, createdValue, username, municipalityValidation)
-      val idUpdated = updateSpeedLimitValue(id, existingValue, username, municipalityValidation).get
-      Seq(loadSpeedLimit(idUpdated).get, loadSpeedLimit(newId).get)
+    val (newId ,idUpdated) = withDynTransaction {
+      (dao.splitSpeedLimit(id, splitMeasure, createdValue, username, municipalityValidation) ,
+      updateSpeedLimit(id, existingValue, username, None, None, municipalityValidation).get)
+    }
+      Seq(getSpeedLimitById(idUpdated).get, getSpeedLimitById(newId).get)
```

Comment:
We can replace this by getSpeedLimitByIds(Seq(idUpdated, newId)) and reduce the database access